#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

gen.add("mask_size", int_t, 0, "", 2, 0, 10)
gen.add("max_variance", double_t, 0, "", 0.1, 0, 1.0)
algo_enum = gen.enum([gen.const("average_variance", str_t, "average_variance", "average_variance"),
                      gen.const("average_distance", str_t, "average_distance", "average_distance")],
                      "smooth_method")
gen.add("smooth_method", str_t, 0, "", "average_variance", edit_method=algo_enum)
gen.add("use_bilateral", bool_t, 0, "", False)
gen.add("bilateral_filter_size", int_t, 0, "", 5, 1, 50)
gen.add("bilateral_sigma_color", double_t, 0, "", 0.04, 0.0, 1.0)
gen.add("bilateral_sigma_space", double_t, 0, "", 5, 0.0, 100)

exit (gen.generate (PACKAGE, "jsk_pcl_ros", "HeightmapMorphologicalFiltering"))
